### PR TITLE
Add PQ vpn feature nat-lab tests

### PR DESCRIPTION
**Pull request by matszczygiel**
### Problem
There are no automatic tests of the PQ feature

### Solution
Add nat-lab tests

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: nat-lab/tests/test_pq.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    async with Ping(client_conn, config.PHOTO_ALBUM_IP).run() as ping:
+        await testing.wait_long(ping.wait_for_next_ping())
+
+    ip = await testing.wait_long(stun.get(client_conn, config.STUN_SERVER))
+    assert ip == wg_server["ipv4"], f"wrong public IP when connected to VPN {ip}"
+
+
+# TODO(LLT-4979): Enable boringtun adapters. Currently, disabled enabled because of a bug inside of boringtun.
+#                 In short, boringtun panics when changing private key of the adapter. It's fixed in the upstream
+#                 but we fear rebasing because it would require extensive testingk
```
Comment (LukasPukenis) [inline nat-lab/tests/test_pq.py:35]: hey, do we have a ticket for that?

**Review by LukasPukenis — COMMENTED** [review]
Nice! +1.5

File: nat-lab/tests/test_pq.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        #         connection_tracker_config=generate_connection_tracker_config(
+        #             ConnectionTag.MAC_VM,
+        #             stun_limits=ConnectionLimits(1, 1),
+        #             nlx_1_limits=ConnectionLimits(2, 2),
+        #         ),
+        #         is_meshnet=False,
+        #     ),
+        #     "10.0.254.7",
+        #     marks=pytest.mark.mac,
+        # ),
```
[the remark this answers is not in the document]
Comment (packgron) [reply nat-lab/tests/test_pq.py:111]: Mac only support boringtun

**Review by packgron — COMMENTED** [review]
+1.0
We need to queue boring tun rebase for v5.0.0
And don't forget to cleanup commit messages!

File: nat-lab/tests/test_pq.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matszczygiel) [reply nat-lab/tests/test_pq.py:111]: Yeah, and boringtun is buggy at the moment

File: nat-lab/tests/test_pq.py (hunk 1 above)
Comment (matszczygiel) [reply nat-lab/tests/test_pq.py:35]: Yeah, LLT-4979

**Comment by matszczygiel** [comment]
@packgron what's wrong with commit messages?

File: nat-lab/tests/test_pq.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        str(wg_server["public_key"]),
+        pq=True,
+    )
+
+    async with Ping(client_conn, config.PHOTO_ALBUM_IP).run() as ping:
+        await testing.wait_long(ping.wait_for_next_ping())
+
+    ip = await testing.wait_long(stun.get(client_conn, config.STUN_SERVER))
+    assert ip == wg_server["ipv4"], f"wrong public IP when connected to VPN {ip}"
+
```
[the remark this answers is not in the document]
Comment (matszczygiel) [reply nat-lab/tests/test_pq.py]: Done

**Review by LukasPukenis — DISMISSED** [review]

**Comment by packgron** [comment]
> @packgron what's wrong with commit messages?

I don't like 'update changelog type comment's' and prefer it squashed with implementation.

**Comment by gytsto** [comment]
Just curious, why not just update `test_vpn_connection` with `pq: bool` parametrization, since test is basically the same

**Comment by matszczygiel** [comment]
@gytsto just to not clutter the VPN tests with branches and stuff further

**Pull request merged**